src/dope-node/communication/communication_handler.go

```diff
@@ -0,0 +1,126 @@
+package communication
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/gorilla/websocket"
+)
+
+var knownAddresses = []string{}
+
+func RegisterEndpoints(boostrapServerAddress string, isBoostrapServer bool, port int) {
+	http.HandleFunc("/ws", wsHandler)
+	log.Println("Running server")
+	if isBoostrapServer {
+		http.HandleFunc("/addresses", httpHandler)
+	} else {
+		// This sends 'hello' message to all known nodes
+		addresses := fetchNodeAddresses(boostrapServerAddress, "127.0.0.1:"+strconv.Itoa(port))
+		for _, addr := range addresses {
+			u := url.URL{Scheme: "ws", Host: addr, Path: "/ws"}
+			conn, _, err := websocket.DefaultDialer.Dial(u.String(), nil)
+			if err != nil {
+				log.Printf("Failed to connect to %s: %v", addr, err)
+				continue
+			}
+			defer conn.Close()
+
+			messageToSend, _ := json.Marshal(Message{Content: "hello"})
+			sendMessage(*conn, messageToSend)
```

Comment:
Please do not ignore errors unless you have a VERRRRY good reason to do so - we usually throw a lot in go, unlike other non error-as-value languages